### ISSUE TITLE
Va-Api support docker

### ISSUE THREAD

**Issue by rafalohaki** [owncast/owncast#2025]
Is there docker confiuguration for enabling Va-Api encoding? couldnt find one
Im using docker on windows with and wanted to enable gpu support

**Comment by gabek** [comment]
Getting vaapi to work in a Docker container would be the same requirements as getting it to work on any other Linux machine. But there's no Owncast image that has that configuration, you'll have to build your own custom image that does what you need. Also, I'm not sure how you'd expose your GPU from Windows to your container. On Linux it seems like it would be straightforward, but I have no idea how these things work in Windows.

You might just have to read up on how to support this kind of thing, specifically with Windows.

**Comment by dumbasPL** [comment]
Is there a reason as to why this isn't supported out of the box on the official images? If size is a concern we could maybe have it under a different tag like `:intel` or `:nvidia` for example.

**Comment by gabek** [comment]
It's simply because not enough people take advantage of hardware accelerated encoding for me to work on building support for it into the Docker images we provide.

**Comment by dumbasPL** [comment]
> It's simply because not enough people take advantage of hardware accelerated encoding 

I think we have a chicken and and egg problem here. People aren't using It because support for it is poor. I wouldn't mind contributing a PR for va-api in docker when i have some time. Are you guys even interested?

Edit: also maybe part of the reason why people don't even attempt to us is is because the documentation for it is poor.

From the docs:
> This should be viewed as an advanced topic that may require a substantial investment in time to get working.

Well, if we had docker support for it then it's as simple as adding `--device=/dev/dri:/dev/dri` to the container and flicking a switch in the web ui.

As for the nvidia side, that can be integrated relatively easily as well.

**Comment by gabek** [comment]
I understand it's frustrating for you to not have this out of the box, but most people run Owncast on VPS hosting that doesn't have access to encoding hardware, that's why it's not used.

Owncast does have support for vaapi, just not in the supplied Docker image. If I had to guess the number of people who have access to encoding hardware _and_ want to use it with Docker (not everyone uses Docker. Maybe 30% of users if I had to guess?) are a very few number of people.

However, if you wanted to add support for it in the Docker image there's not a lot of downsides other than size, as long as support for it isn't something that I'd be left with down the road as I have no way to test it. We deploy test images every day for testing Owncast, but there's no environment that has access to hardware accelerated encoding. In general most people don't.
